### PR TITLE
Be da 0410

### PR DESCRIPTION
**Pull request by habibdevsy**
- Merge with branch 'BE-DA-0409'
- create subscription by admin
- extra Subscription For Day By Admin

### CODE REVIEW

File: backend/config/validator/validation.yaml [hunk 1]
```diff
@@ -653,3 +653,15 @@ App\Request\Admin\Order\OrderStateUpdateByAdminRequest:
...
             - NotBlank: ~
+
+App\Request\Admin\Subscription\AdminCreateStoreSubscriptionRequest:
+    properties:
+        storeProfileId:
+            - NotBlank: ~
+        packageId:
+            - NotBlank: ~
+
+App\Request\Admin\Subscription\AdminExtraSubscriptionForDayptionRequest:
```
Comment (ralsalloum) [inline backend/config/validator/validation.yaml]: request class name must be fixed.

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php [hunk 2]
```diff
@@ -99,4 +113,183 @@ public function deleteAllStoreFutureSubscriptionsByStoreOwnerId(int $storeOwnerI
...
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="create subscription",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php]: request body has no Data, msg, and status code. :)

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php [hunk 3]
```diff
@@ -99,4 +113,183 @@ public function deleteAllStoreFutureSubscriptionsByStoreOwnerId(int $storeOwnerI
...
+     *      description="Return error.",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code", description="9351 or 9157"),
+     *          @OA\Property(type="string", property="msg", description="package not exist or store owner profile not exist!"),
+     *        )
+     *     )
+     * 
+     * @Security(name="Bearer")
+     */
+    public function createSubscription(Request $request): JsonResponse
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php:167]: We may use another function name in order to differentiate it form createSubscription function that used by store owner.

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php [hunk 4]
```diff
@@ -99,4 +113,183 @@ public function deleteAllStoreFutureSubscriptionsByStoreOwnerId(int $storeOwnerI
...
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="create subscription For Day",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
```
Comment (ralsalloum) [inline backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php]: request body has no Data, msg, and status code. :)

**Review by ralsalloum — CHANGES_REQUESTED** [review]
please resolve the requested changes.

File: backend/config/validator/validation.yaml (hunk 1 above)
Comment (habibdevsy) [reply backend/config/validator/validation.yaml]: done

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php (hunk 2 above)
Comment (habibdevsy) [reply backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php]: this is new style ^_^.
done

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php (hunk 3 above)
Comment (habibdevsy) [reply backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php:167]: This function will only be called by (AdminStoreSubscriptionService)

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php (hunk 4 above)
Comment (habibdevsy) [reply backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php]: done

File: backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php (hunk 3 above)
Comment (ralsalloum) [reply backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php:167]: How a controller function is being called by a service one? :)

**Review by ralsalloum — APPROVED** [review]

**Pull request closed without merging**